docs/code/dns/main.c

```diff
@@ -10,7 +10,7 @@ void alloc_buffer(uv_handle_t *handle, size_t suggested_size, uv_buf_t *buf) {
   buf->len = suggested_size;
 }
 
-void on_read(uv_stream_t *client, ssize_t nread, const uv_buf_t *buf) {
+void on_read(uv_stream_t *client, uv_ssize_t nread, const uv_buf_t *buf) {
```

Comment:
What about:
size_t nread
if(nread!=0) -> uv_buf_t is valid
else -> check uv_stream_t for errors.

Reply:
This patch is about using `uv_ssize_t` instead of `uv_ssize_t`. I think discussion of logic is out of scope of this change.

Reply:
Agreed. @bwijen can you please create a new issue / PR with what you spotted? Thanks!

Reply:
Agreed: #1548